docker-compose.yml

```diff
@@ -0,0 +1,11 @@
+version: '3.8'
+
+services:
+  app:
+    build: .
+    image: ocw_oer_export
+    entrypoint: ["python", "-m", "ocw_oer_export.cli"]
+    command: ["--create_csv"]
+    tty: true
+    volumes:
+      - ./output:/output
```

Comment:
When I ran this, the `output` folder that was created was owned by `root`. The script then failed to save any files in the folder with:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/src/ocw_oer_export/cli.py", line 38, in <module>
    main()
  File "/src/ocw_oer_export/cli.py", line 30, in main
    create_csv(source=args.source)
  File "/src/ocw_oer_export/create_csv.py", line 190, in create_csv
    with open(output_path, "w", newline="", encoding="utf-8") as csv_file:
PermissionError: [Errno 13] Permission denied: '/output/ocw_oer_export.csv'
```

This is a fairly common issue with Docker. I was able to `chown` the folder to be owned by my current user, then run the commands. What I'd recommend here is creating a `private` folder, creating an `output` folder within that, committing both folders to git, then adding `private` to `.gitignore` after it's committed. You'll then need to change the output paths to `./private/output/`. This will ensure the folder is created as the current user when the repo is cloned, avoiding this issue. Placing the `output` folder in an ignored folder called `private` allows the possibility of adding other gitignored folders in the future if necessary.

Reply:
I committed the directories, using a `.gitkeep` and then removing it later after adding the directory to `.gitignore`. See if it works now?

Reply:
You can't remove the .gitkeep, because then the directory is empty, and git won't keep it. I'd add a comment in the gitkeep file, too.